### ISSUE TITLE
How to deal with `RngBoost`

### ISSUE THREAD

**Issue by dmcdougall** [libqueso/queso#455]
We have an RNG interface and at present we can select between gsl and boost, although I'm not sure if the boost interface is functional.

Since we're moving towards making boost an optional dependency (requiring it only when the compiler is old enough), I thought I'd open this ticket to discuss what we can do about the random number generation.

When the compiler is new enough, of course we can use the random number generators in the C++11 standard library.  If it isn't new enough, should we fall back to boost or use something else?

**Comment by pbauman** [comment]
IMHO, I'd say get rid of the Boost one, particularly since the C++11 one has been added. Then, if the user doesn't have a new enough compiler (anyone on RHEL/CentOS 7 does), they can fall back to GSL. Just my two cents.

**Comment by roystgnr** [comment]
Fall back on GSL, but when GSL's not there fall back on Boost.  We're trying to get to "User needs C++11 or Boost but doesn't need both" in the long run, with GSL purely optional for licensing reasons.

**Comment by pbauman** [comment]
Fair enough. Was just hoping we could strip away Boost entirely and not have maintain it since it's been such a PITA.

**Comment by dmcdougall** [comment]
You will be able to if you're compiler is new enough.

**Comment by pbauman** [comment]
> You will be able to if you're compiler is new enough.

Right, I understand. I was just hoping to be able to reduce the QUESO maintenance burden by stripping out Boost entirely. But the licensing point is fair, particularly since QUESO is not all-in C++11 yet.

**Comment by dmcdougall** [comment]
> Right, I understand. I was just hoping to be able to reduce the QUESO maintenance burden by stripping out Boost entirely. But the licensing point is fair, particularly since QUESO is not all-in C++11 yet.

The licensing point should actually be called a licensing 'uncertainty'.  It's not that GSL's license is prohibitive, it's that none of us know it's prohibitive because none of us are lawyers.
